### DISCUSSION TITLE
Support DropTarget Size Options

### DISCUSSION THREAD

**Discussion by garyvh2** [mathuo/dockview#427] [Ideas]
Hey there @mathuo.

Our team is running into an issue with the size of the Drop Target Area, as `Top`, `Right`, `Left` and `Bottom` applied on `DockviewComponent` is too small, we wanted to see if there would be a possibility to add the quadrant's overlayModel `activationSize` and `size` as possible options to the library or maybe the `API`, in order to customize the size of the target zones, we could even add the `acceptedTargetZones` so we can customize which overlays would be possible.

![image](https://github.com/mathuo/dockview/assets/21265830/e4541ae0-976a-40fd-a7b8-8fa4253923fd)

**Comment by mathuo** [comment]
Would it be suitable to expose `activationSize/size` at a `DockviewReact` component prop level so that it applies for everything with that component or are you talking more about custom values per tab group?

**Reply by garyvh2** [reply]
Is mostly for the `DockviewComponent`

This is how the `Top`, `Right`, `Left` and `Bottom` at `DockviewComponent` looks like now

![image](https://github.com/mathuo/dockview/assets/21265830/9a636c86-bdc3-4ec5-bb77-31638708e8d5)

This is increasing the `activationSize to 25` and the `size to 50`

![image](https://github.com/mathuo/dockview/assets/21265830/e62bf652-8f2d-4d0b-9047-336491d943da)

----

For this example we are changing it directly at [`dockviewComponent.ts ln.465-468` ](https://github.com/mathuo/dockview/blob/master/packages/dockview-core/src/dockview/dockviewComponent.ts#L465-L468)

```javascript
dropTarget.options.overlayModel.size.value = 50
dropTarget.options.overlayModel.activationSize.value = 25
```

--- 

To answer you question, in this case we only care about the global dock container, but it would be a good feature if this could be added to the API, so it can be customized on panels specifically.

**Reply by mathuo** [reply]
@garyvh2 I will get something added in that allows this to be configurable. Currently just in the process of releasing a new version (1.9.0) but after that is done I will add this change in as a minor update (1.9.1 ish) since the change should be trival.

**Comment by mathuo** [comment]
Released in 1.9.1 for the far edge overlay https://dockview.dev/docs/components/dockview#drag-and-drop

**Comment by mathuo** [comment]
https://dockview.dev/docs/core/dnd/dragAndDrop
